addons/tagging/functions/fnc_quickTag.sqf

```diff
@@ -22,19 +22,22 @@ if (GVAR(quickTag) == 0) exitWith {};
 params ["_unit"];
 
 private _possibleTags = [];
+private _useRandom = false;
 
 // Last Used
 if (GVAR(quickTag) == 1) then {
     private _lastUsedTagClass = _unit getVariable [QGVAR(lastUsedTag), nil];
 
-    if (!isNil "_lastUsedTagClass") then {
+    if (isNil "_lastUsedTagClass") then {
+        _useRandom = true;
+    } else {
         private _lastUsedTag = GVAR(cachedTags) select {(_x select 0) == _lastUsedTagClass};
         _possibleTags = _lastUsedTag;
     };
 };
 
 // Random X
-if (GVAR(quickTag == 2)) then {
+if ((GVAR(quickTag) == 2) || _useRandom) then {
```

Comment:
Lazy eval?

Reply:
Might be not worth it for plain booleans.

Reply:
That's actually a good point, I know it's not always worth it especially with `&&`, but that would be interesting to test. And we should really add this more specifically to our guidelines.

Reply:
it's mentioned at [Code Optimisation](https://community.bistudio.com/wiki/Code_Optimisation#Conditions)

Reply:
Yes, I meant our wiki.